Assignment4/Assignment4.py

```diff
@@ -0,0 +1,246 @@
+            logistic_regression_model = statsmodels.api.Logit(Y, predict)
+            logistic_regression_model_fitted = logistic_regression_model.fit()
+            t = round(logistic_regression_model_fitted.tvalues[1], 6)
+            p = "{:.6e}".format(logistic_regression_model_fitted.pvalues[1])
+
+            # appending the p and t values to the table
+            t_score.append(t)
+            p_value.append(p)
+
+            # calculatin the random forest variable importance features
+            importance = model.feature_importances_
+            value = importance[i]
+
+            # appending the random forest variable to the table
+            RF_VarImp.append(value)
+
+            Plot.append("NA")
+            MWR_Unweighted.append("NA")
+            MWR_Weighted.append("NA")
+
+if len(pd.unique(Y1)) > 2:
+
+    for i in range(0, X.shape[1]):
+        Response.append(R_columns[0])
+
+        n = len(pd.unique(X.iloc[:, i]))
+        if n <= 4:
+
+            # appending the column name along with its type
+            Predictor.append(P_Columns[i] + '(Categorical)')
+
+            # calling the plot function of this type
+            cont_resp_cat_predictor()
+
+            t_score.append("NA")
+            Plot.append("NA")
+            p_value.append("NA")
+            RF_VarImp.append("NA")
+            MWR_Unweighted.append("NA")
+            MWR_Weighted.append("NA")
+
+        else:
+
+            # appending the column name along with its type
+            Predictor.append(P_Columns[i] + '(Continuous)')
+
+            # calling the plot function
+            cont_response_cont_predictor()
+
+            # p and t value calculation using logistic regression method
+            predict = statsmodels.api.add_constant(X.iloc[:, i])
+            linear_regression_model = statsmodels.api.OLS(Y, predict)
+            linear_regression_model_fitted = linear_regression_model.fit()
+            t = round(linear_regression_model_fitted.tvalues[1], 6)
+            p = "{:.6e}".format(linear_regression_model_fitted.pvalues[1])
+
+            # appending the p and t values to the table
+            t_score.append(t)
+            p_value.append(p)
+
+            importance = model.feature_importances_
+            values = importance[i]
+            # appending the random forest variable to the table
+            RF_VarImp.append(values)
+
+            Plot.append("NA")
+            MWR_Unweighted.append("NA")
+            MWR_Weighted.append("NA")
+
+# storing all the results in a data frame and showing
+
+if len(pd.unique(Y1)) < 2:
+    Summary_Table['Response(Boolean)'] = Response
+else:
+    Summary_Table['Response(Continuous)'] = Response
+
+Summary_Table['Predictor'] = Predictor
+Summary_Table['Plot'] = Plot
+Summary_Table['p-value'] = p_value
+Summary_Table['t-score'] = t_score
+Summary_Table['RF VarImp'] = RF_VarImp
+Summary_Table['MWR Unweighted'] = MWR_Unweighted
+Summary_Table['MWR Weighted'] = MWR_Weighted
+
+Summary_Table
```

Comment:
Your code is not running and a lot of functionalities don't work and have a lot of errors.